tests/pytest/src/global_register/global_register.py

```diff
@@ -0,0 +1,62 @@
+import os
+import uuid
+import random
+
+class GlobalRegister(object):
+    def __init__(self):
+        self.registered_ports = []
+        self.registered_files = []
+        # contains dynamic or private ports that cannot be registered with IANA
+        self.min_port = 49152
+        self.max_port = 65535
+        self.global_prefix = "pytest"
+
+    def get_uniq_filename(self, prefix, extension="txt", subdir="/tmp"):
+        unique_file_name = None
+        while True:
+            unique_file_name = os.path.join(subdir,"%s_%s_%s.%s" % (self.global_prefix, prefix, str(uuid.uuid4()), extension))
+            if not self.is_file_name_registered(unique_file_name):
+                self.register_file_name(unique_file_name)
+                break
+        return unique_file_name
+
+    # def get_uniq_dirname(self):
+    #     pass
+
+    def get_uniq_port(self):
+        random_port = None
+        while True:
+            if self.is_available_port():
+                random_port = random.randint(self.min_port, self.max_port)
+                if not self.is_port_registered(random_port):
+                    self.register_port(random_port)
+                    break
+            else:
+                break
+        return random_port
+
+    def is_port_registered(self, port):
+        return port in self.registered_ports
```

Comment:
It would be good to check whether the port is in use by the system at the moment. Also, it would be practical to bind the port here instead of the caller.